src/components/templates/Camp/hooks/usePartygoersOverlay.tsx

```diff
@@ -35,37 +32,33 @@ export const usePartygoersOverlay: ReactHook<
   partygoers,
   setSelectedUserProfile,
 }) => {
-  return useMemo(
-    () =>
-      // @debt this can be undefined because our types are broken so check explicitly
-      showGrid ? (
-        partygoers?.map(
-          (partygoer) =>
-            partygoer?.id && ( // @debt workaround, sometimes partygoers are duplicated but the new ones don't have id's
-              <MapPartygoerOverlay
-                key={partygoer.id}
-                partygoer={partygoer}
-                venueId={venueId}
-                myUserUid={userUid ?? ""} // @debt fix this to be less hacky
-                totalRows={rows}
-                totalColumns={columns}
-                withMiniAvatars={withMiniAvatars}
-                setSelectedUserProfile={setSelectedUserProfile}
-              />
-            )
-        )
-      ) : (
-        <div />
-      ),
-    [
-      showGrid,
-      partygoers,
-      venueId,
-      userUid,
-      rows,
-      columns,
-      withMiniAvatars,
-      setSelectedUserProfile,
-    ]
-  );
+  return useMemo(() => {
+    if (!showGrid) return null;
+    // @debt this can be undefined because our types are broken so check explicitly
+    return partygoers.map((partygoer) => {
+      if (!partygoer.id || !partygoer.data?.[venueId]) return null;
+
+      return (
+        <MapPartygoerOverlay
+          key={partygoer.id}
+          partygoer={partygoer}
+          position={partygoer?.data?.[venueId]}
+          myUserUid={userUid} // @debt fix this to be less hacky
```

Comment:
I think the `@debt` comment here related to `myUserId` having a default instead of passing through as `undefined`, so I think we can remove that comment with the change here.

See comment earlier RE: `position`, I think it's better to just pass `row` and `column` down as individual props. So we should probably extract them in a `const` higher up.